theseus/utils/examples/motion_planning/motion_planner.py

```diff
@@ -171,6 +172,9 @@ def __init__(
                     )
                 )
             )
+        # Add any extra user-provided cost functions before creating optimizer
+        if obj_modifier_callback is not None:
```

Comment:
Looks good. Agreed on merging after adding some unit test.

Can you briefly explain the intended behavior for when the callback is triggered?

Reply:
I realized later that none of this code has any unit tests in place, since this is mostly meant to be taken as examples of how to use the library. The add on in this PR seems to be working well on our private experiment repo, so I propose merging as is. 

For the explanation, do you mean in the comments? Or here?

Reply:
Sounds good on the tests in that case.

For explanation, actually, yes both :)

Reply:
This lets users add other cost functions to the motion planner or even remove some of the default ones. We need to do it this way because the optimizer gets created inside the MotionPlanner, which locks the objective from allowing further changes.

Reply:
Ah I see. Hmm another alternative could be that a MotionPlannerObjective class gives you a base objective you can modify as you see fit. Then you can pass to a wrapper class MotionPlannerLayer that gives you back a TheseusLayer version of your modified objective. Basically splitting the current functionality in to two steps.

Reply:
I like that suggestion, I think it's cleaner. Let's go with that.

Reply:
See #277